### PR TITLE
[BUG] ExecuteMonitor inserting metadata doc during dry run

### PR DESCRIPTION
**Pull request by petardz**
*Issue #, if available:*

*Description of changes:*

Fixed ExecuteMonitor Action inserting metadata doc during dryrun

*CheckList:*
[ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt [hunk 1]
```diff
@@ -130,7 +130,7 @@ class TransportExecuteMonitorAction @Inject constructor(
                                 docLevelMonitorQueries.initDocLevelQueryIndex(monitor.dataSources)
                                 log.info("Central Percolation index ${ScheduledJob.DOC_LEVEL_QUERIES_INDEX} created")
                             }
-                            val (metadata, _) = MonitorMetadataService.getOrCreateMetadata(monitor)
+                            val (metadata, _) = MonitorMetadataService.getOrCreateMetadata(monitor, skipIndex = true)
```
Comment (lezzago) [inline alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt:133]: This should only be true if the dryrun param is true

File: alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt [hunk 2]
```diff
@@ -339,7 +342,9 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
                     // If we reached limit for total number of fields in mappings after rollover
                     // it means that source index has more then (FIELD_LIMIT - 3) fields (every query index has 3 fields defined)
                     // TODO maybe split queries/mappings between multiple query indices?
-                    if (e.message?.contains("Limit of total fields") == true) {
+                    val unwrappedException = ExceptionsHelper.unwrapCause(e) as Exception
+                    log.debug("exception after rollover queryIndex index: $targetQueryIndex exception: ${unwrappedException.message}")
+                    if (unwrappedException.message?.contains("Limit of total fields") == true) {
```
Comment (lezzago) [inline alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt:347]: Will this be changed to false from true ever?

**Review by lezzago — CHANGES_REQUESTED** [review]

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt (hunk 1 above)
Comment (petardz) [reply alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt:133]: even if monitorId is null ?

Comment (petardz) [reply alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt:133]: I think if monitorId is null we definitely can't save metadata (there's no monitor to tie it to). 

Dryrun can be true for existing monitor and we take that into account before indexing metadata inside DocumentLevelMonitorRunner

Comment (lezzago) [reply alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt:133]: Oh I missed that part of the logic, good call out.

File: alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt [hunk 3]
```diff
@@ -341,6 +342,47 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
         assertEquals("Didn't match all 4 queries", 1, findings[0].docLevelQueries.size)
     }
 
+    fun `test execute monitor without create when no monitors exists`() {
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt:345]: Lets have one more test where the monitor is created and dryrun is true and there was not update to the metadata doc

Comment (lezzago) [reply alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt:345]: Not actually valid to have a test like that.

**Review by lezzago — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**